filechooser-portal/FileChooserDialog.vala

```diff
@@ -355,6 +354,13 @@ public class Files.FileChooserDialog : Hdy.Window, Xdp.Request {
             stack.notify["visible-child"].connect (() => {
                 revealer.reveal_child = stack.visible_child_name == "search";
             });
+
+            if (action == Gtk.FileChooserAction.SAVE) {
+                entry.set_placeholder_text (_("Enter new filename"));
+                entry.grab_focus ();
+            } else {
+                tree_view.grab_focus ();
+            }
```

Comment:
this don't need to be done during realize, Gtk don't seem to do any modification in the placeholder and the focus widget after construction (like it does with the search entry).

`entry.placeholder_text` can be set when we move it to the action bar. The focus grab is better placed in the construct method itself, since it aren't dealing with modifications in the internal functionality of the chooser widget.

Reply:
Yes, now the other grab focus calls are removed you are right.  Will fix.

Reply:
Done